### PR TITLE
LUCENE-9760: Hunspell: print total memory usage in TestAllDictionarie…

### PR DESCRIPTION
**Pull request by donnerpeter**
…s, cleanup

# Description

We'd like Hunspell dictionaries not to take too much memory, especially when there's many of them loaded

# Solution

Print the memory usage of all known dictionaries together

# Tests

A test-only change

# Checklist

Please review the following and check all that apply:

- [x] I have reviewed the guidelines for [How to Contribute](https://wiki.apache.org/solr/HowToContribute) and my code conforms to the standards described there to the best of my ability.
- [x] I have created a Jira issue and added the issue ID to my pull request title.
- [x] I have given Solr maintainers [access](https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to contribute to my PR branch. (optional but recommended)
- [x] I have developed this patch against the `master` branch.
- [x] I have run `./gradlew check`.
- [ ] I have added tests for my changes.
- [ ] I have added documentation for the [Ref Guide](https://github.com/apache/lucene-solr/tree/master/solr/solr-ref-guide) (for Solr changes only).

### CODE REVIEW

File: lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java [hunk 1]
```diff
@@ -181,11 +188,16 @@ public void testDictionariesLoadSuccessfully() throws Exception {
       }
     } finally {
       executor.shutdown();
-      executor.awaitTermination(1, TimeUnit.MINUTES);
+      assertTrue(executor.awaitTermination(1, TimeUnit.MINUTES));
```
Comment (dweiss) [inline lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java:191]: The test runner would also capture and report any escaped threads, even if this falls through.

Comment (donnerpeter) [reply lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java:191]: This was yellow-highlighted by the IDE, and it was easier to surrender :)

Comment (dweiss) [reply lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java:191]: Use vim? :)

Comment (donnerpeter) [reply lucene/analysis/common/src/test/org/apache/lucene/analysis/hunspell/TestAllDictionaries.java:191]: Oh. I'm too addicted to IntelliJ and its warnings (especially having created some of them) :)

**Pull request merged**